src/compile/index.ts

```diff
@@ -74,7 +79,7 @@ function get_name(filename) {
 }
 
 export default function compile(source: string, options: CompileOptions = {}) {
-	options = assign({ generate: 'dom', dev: false }, options);
+	options = assign({ generate: 'dom', dev: false, shadowDom: options.customElement }, options);
```

Comment:
This defaults `shadowDom` to whatever `customElement` may be set to. So if it's `true`, `shadowDom` will be `true` by default (which is the behavior we want for backwards compatibility).